### PR TITLE
add collection publishing roles

### PR DESCRIPTION
**Pull request by gtema**

### CODE REVIEW

**Comment by gtema** [comment]
recheck

**Comment by gtema** [comment]
recheck

**Comment by gtema** [comment]
recheck

**Comment by gtema** [comment]
recheck

**Comment by gtema** [comment]
recheck

**Comment by gtema** [comment]
recheck

**Comment by gtema** [comment]
recheck

**Comment by gtema** [comment]
regate

File: linters-requirements.txt [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# linters have different requirements than test ones, some would
+# conflict, like ansible version required by ansible-lint.
+flake8
+yamllint>=1.23.0
+ansible-lint>=4.3.0a1,<5
```
Comment (lego963) [inline linters-requirements.txt:5]: Is it ok to use alpha?

Comment (gtema) [reply linters-requirements.txt:5]: currently having same requirements as upstream zuul-jobs, thus - should be fine. Actially 99% of this PR is copying over stuff from zuul-jobs, while the ansible-collections stuff comes from https://review.opendev.org/#/c/730360/ (seems to stuck for a while)

**Comment by gtema** [comment]
regate

**Pull request merged**